### PR TITLE
Batch transactions improvements

### PR DESCRIPTION
**Pull request by CiprianDraghici**
### Feature
1. Standardize send transactions (normal flow, batch transaction)
2. Standardize TransactionSender component
3. Standardize TransactionTracker component
4. Avoid functionality collision between transactions tracking components (for normal flow and batch transactions)
5. Simplify batch transaction usage. This can be enabled using the dedicated `sendBatchTransactions` function
6. Avoid sending transactions when `signTransactions` function is called
### Contains breaking changes
[x] No

[] Yes

### Updated CHANGELOG
[x] Yes

### Testing
[x] User testing
[x] Unit tests

### CODE REVIEW

File: src/apiCalls/transactions/sendSignedBatchTransactions.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+}: SendBatchTransactionsPropsType) {
+  const { apiAddress, apiTimeout } = networkSelector(store.getState());
+
+  try {
+    const batchId = buildBatchId({
+      sessionId,
+      address
+    });
+
+    const payload = {
```
Comment (arhtudormorar) [inline src/apiCalls/transactions/sendSignedBatchTransactions.ts:34]: add type here

File: src/components/TransactionSender/TransactionSender.ts [hunk 2]
```diff
@@ -22,126 +18,119 @@ import {
...
-        sendingRef.current = true;
-        const transactionsToSend = transactions.map((tx) => newTransaction(tx));
-        const responseHashes = await sendSignedTransactionsAsync(
-          transactionsToSend
-        );
-
-        const newStatus = TransactionServerStatusesEnum.pending;
-        const newTransactions = transactions.map((transaction) => {
-          if (responseHashes.includes(transaction.hash)) {
-            return { ...transaction, status: newStatus };
```
Comment (arhtudormorar) [inline src/components/TransactionSender/TransactionSender.ts:96]: I think this was correct - maybe bring back?

File: src/components/TransactionSender/TransactionSender.ts [hunk 3]
```diff
@@ -22,126 +18,119 @@ import {
...
-        removeTransactionParamsFromUrl({
-          transaction
-        });
-      } catch (error) {
-        const errorMessage =
-          (error as AxiosError).response?.data?.message ??
-          (error as any).message;
-
-        dispatch(
-          updateSignedTransactions({
```
Comment (arhtudormorar) [inline src/components/TransactionSender/TransactionSender.ts:129]: I think this was correct

File: src/wrappers/DappProvider/CustomComponents.tsx [hunk 4]
```diff
@@ -21,50 +18,32 @@ export interface CustomComponentsType {
...
+      <TxSender {...transactionSender?.props} />
+      <TxTracker {...transactionTracker?.props} />
 
-      {Boolean(enableBatchTransactions) && (
-        <>
-          <BatchTxsSender {...batchTransactionsSender?.props} />
-          <BatchTxsTracker {...batchTransactionsTracker?.props} />
-        </>
-      )}
+      <BatchTxsTracker {...batchTransactionsTracker?.props} />
```
Comment (arhtudormorar) [inline src/wrappers/DappProvider/CustomComponents.tsx:46]: I think this should be integrated by default in TxTracker

File: src/components/TransactionSender/TransactionSender.ts (hunk 2 above)
Comment (CiprianDraghici) [reply src/components/TransactionSender/TransactionSender.ts:96]: This is just refactored, not removed. The functionality is still there

File: src/wrappers/DappProvider/CustomComponents.tsx (hunk 4 above)
Comment (CiprianDraghici) [reply src/wrappers/DappProvider/CustomComponents.tsx:46]: For the moment we can't. The props necessary for these components are totally different. We can't mix them.

**Review by ioandanpopa — DISMISSED** [review]

File: src/apiCalls/transactions/sendSignedBatchTransactions.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      sessionId,
+      address
+    });
+
+    const payload: BatchTransactionsRequestType = {
+      transactions,
+      id: batchId
+    };
+
+    const response = await axios.post<BatchTransactionsResponseType>(
```
Comment (arhtudormorar) [inline src/apiCalls/transactions/sendSignedBatchTransactions.ts:44]: you can put this in async actions

File: src/services/transactions/updateSignedTransactions.ts [hunk 6]
```diff
@@ -33,3 +35,10 @@ export function setTransactionsDisplayInfoState(
 ) {
   store.dispatch(setTransactionsDisplayInfo(payload));
 }
+
+export function updateSignedTransactionCustomTransactionInformationState(payload: {
```
Comment (arhtudormorar) [inline src/services/transactions/updateSignedTransactions.ts:39]: I think it would be better if this was a hook

Comment (CiprianDraghici) [reply src/services/transactions/updateSignedTransactions.ts:39]: Why ? This is just an utils function that updates a specific field in the store

**Review by ioandanpopa — APPROVED** [review]

**Pull request merged**